### PR TITLE
Feature/10560 push changes

### PR DESCRIPTION
**Pull request by MarianoTucat**
updated docs for webhooks

### CODE REVIEW

File: docs/specs/tripgo.swagger.yaml [hunk 1]
```diff
@@ -835,7 +835,7 @@ paths:
         required: true
         description: Identifier of trip as returned by previous server call.
 
-  /trip/{id}/save:
+  /trip/save/{id}:
```
Comment (nighthawk) [inline docs/specs/tripgo.swagger.yaml]: Out of curiousity, what's the reason for this change? I think the old style was more RESTy.

File: docs/enterprise/tripurls.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+### Receiving notifications
+
+The `hookURL` can be used to register a callback or web-hook, and our platform will then inform if any change in that trip occurs by POSTing to the registered url the `tripID` along with the `tripURL` for the client to retrieve the updated trip.
+
+The expected flow in this case is the following one:
+
+1. A trip is computed and our platform has sources of information for real-time updates for it, and therefore, the `hookURL` is returned.
+2. The client that wants to receive notifications about real-time changes does a POST to the `hookURL` ([docs](/specs/#tag/Trips%2Fpaths%2F~1trip~1hook~1%7Bid%7D%2Fpost)) send the webhook as `url` and any required header (starting with `x-`) as body of the request and obtains a 204 response.
+3. At any given time the trip is updated with any real-time change, our platform does a POST to the registered web-hook.
+4. After two hours of not getting any update, the trip is discarded from memory and won't receive any more updates.
```
Comment (nighthawk) [inline docs/enterprise/tripurls.md]: Hmm, but what if I plan a trip for tomorrow morning and want to receive notifications for that? I don't expect any updates overnight, but do expect some in the morning.

I think this would make more sense as: If there are no updates after an hour past the arrival time of the trip.

File: docs/specs/tripgo.swagger.yaml [hunk 3]
```diff
@@ -896,7 +896,91 @@ paths:
...
+        Rather than constructing this URL directly, you should get it from the `hookURL` property from a computed trip.
+      parameters:
+      - name: input
+        in: body
+        schema:
+          type: object
+          properties:
+            url:
+              type: string
+              description: The webhook url to be notified when the trips is changed due to real-time updates. When that happens, our servers will POST to the registered URL a JSON object with the `tripID` and the `tripURL` to be retrieved.
```
Comment (nighthawk) [inline docs/specs/tripgo.swagger.yaml]: Would be nice if it also posted some kind of enum indicating the change, e.g., "TRIP_UPDATED" or "TRIP_CANCELED"

File: docs/enterprise/tripurls.md (hunk 2 above)
Comment (MarianoTucat) [reply docs/enterprise/tripurls.md]: agree, will make that change, and then update the docs

File: docs/specs/tripgo.swagger.yaml (hunk 1 above)
Comment (MarianoTucat) [reply docs/specs/tripgo.swagger.yaml]: This is just to be consistent with our endpoints. Agree that the other way is better, but will create a PR and then update the docs.

File: docs/specs/tripgo.swagger.yaml (hunk 3 above)
Comment (MarianoTucat) [reply docs/specs/tripgo.swagger.yaml]: not sure what you mean by trip_canceled, something like Uber driving cancelling the ride?

Comment (nighthawk) [reply docs/specs/tripgo.swagger.yaml]: Yes, or you planning a train trip and then your train getting canceled.
